files/app/main/status/e/backup-config.ut

```diff
@@ -0,0 +1,51 @@
+{%
+/*
+ * Part of AREDN® -- Used for creating Amateur Radio Emergency Data Networks
+ * Copyright (C) 2024 Tim Wilkinson
+ * See Contributors file for additional contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Additional Terms:
+ *
+ * Additional use restrictions exist on the AREDN® trademark and logo.
+ * See AREDNLicense.txt for more info.
+ *
+ * Attributions to the AREDN® Project must be retained in the source code.
+ * If importing this code into a new or existing project attribution
+ * to the AREDN® project must be added to the source code.
+ *
+ * You must not misrepresent the origin of the material contained within.
+ *
+ * Modified versions must be modified to attribute to the original source
+ * and be marked in reasonable ways as differentiate it from the original
+ * version
+ */
+%}
+{%
+if (request.env.REQUEST_METHOD === "PUT") {
+    response.headers["HX-Redirect"] = request.env.REQUEST_URI;
+    return;
+}
+response.override = true;
+const file = configuration.backup();
+if (!file) {
+    uhttpd.send(`Status: 404 Not Found\r\nCache-Control: no-store\r\n\r\n`);
```

Comment:
Pedantic nit, but.. Is this the right HTTP code?

The server failed to find the file(s), which *should* have existed. A 404 definition states that the resource may be available in the future, and that subsequent requests are permissible. I don't think the availability of the backup files would change?

204 is No Content, which may be more applicable. The server processed the request, but had nothing to return, it also does not guarantee or imply that you could retry and have success.

I suppose this isn't actually a REST API, but.. 🤷

Reply:
The file is missing so I'm happy with a 404 and most people understand what it means. Not that it will ever happen anyway.